pkg/manager/member/tidb_failover.go

```diff
@@ -45,22 +45,22 @@ func (tf *tidbFailover) Failover(tc *v1alpha1.TidbCluster) error {
 		}
 	}
 
-	if tc.Spec.TiDB.MaxFailoverCount != nil {
+	if tc.Spec.TiDB.MaxFailoverCount != nil && *tc.Spec.TiDB.MaxFailoverCount > 0 {
 		maxFailoverCount := *tc.Spec.TiDB.MaxFailoverCount
-		if maxFailoverCount > 0 && len(tc.Status.TiDB.FailureMembers) >= int(maxFailoverCount) {
+		if len(tc.Status.TiDB.FailureMembers) >= int(maxFailoverCount) {
 			klog.Warningf("the failure members count reached the limit:%d", tc.Spec.TiDB.MaxFailoverCount)
 			return nil
 		}
-	}
-	for _, tidbMember := range tc.Status.TiDB.Members {
-		_, exist := tc.Status.TiDB.FailureMembers[tidbMember.Name]
-		deadline := tidbMember.LastTransitionTime.Add(tf.tidbFailoverPeriod)
-		if !tidbMember.Health && time.Now().After(deadline) && !exist {
-			tc.Status.TiDB.FailureMembers[tidbMember.Name] = v1alpha1.TiDBFailureMember{
-				PodName:   tidbMember.Name,
-				CreatedAt: metav1.Now(),
+		for _, tidbMember := range tc.Status.TiDB.Members {
```

Comment:
Why move this for loop in the control of failover? We may still need to update the status for the failure members even if the failover is disabled.

Reply:
yes, the `for _, tidbMember := range tc.Status.TiDB.Members {` loop is part of the status syncing. the user can choose not to failover when a member fails, but the controller should record failure members in the status always.

Reply:
I think there is no updating status here both in the previous or in the current.